services/pull/check.go

```diff
@@ -205,7 +205,7 @@ func handle(data ...queue.Data) {
 		if err != nil {
 			log.Error("GetPullRequestByID[%s]: %v", prID, err)
 			continue
-		} else if pr.Status != models.PullRequestStatusChecking {
+		} else if pr.HasMerged {
```

Comment:
What about status checking?

Reply:
So, it's possible that you can pop the same pr off the queue that you have just worked on (or are working on), implying that at least part way through your processing the pull head has changed. There's only one reason to not process the pr further - it has been merged - if it's been set mergable then that could be already out of date, similarly for conflicted.

has_merged should probably be folded into status tbh.